src/thor/isochrone.cc

```diff
@@ -125,17 +125,20 @@ void Isochrone::ConstructIsoTile(const bool multimodal,
   // Range of grids in longitude space
   float dlon = max_distance / DistanceApproximator<PointLL>::MetersPerLngDegree(center_ll.lat());
 
-  // Optimize for 600 cells in latitude (slightly larger for multimodal).
-  // Round off to nearest 0.001 degree. TODO - revisit min and max grid sizes
-  float grid_size = multimodal ? dlat / 500.0f : dlat / 300.0f;
-  if (grid_size < 0.001f) {
-    grid_size = 0.001f;
-  } else if (grid_size > 0.005f) {
-    grid_size = 0.005f;
-  } else {
-    // Round to nearest 0.001
-    int r = std::round(grid_size * 1000.0f);
-    grid_size = static_cast<float>(r) * 0.001f;
+  float grid_size = api.options().isochrone_grid_size();
```

Comment:
Hm I’ll need to read up a bit on what my past self thought in that issue:)

Intuitively I’m not sure I like an (unbound) request option for such a pretty sensitive parameter. I’d rather like us pick a (much!) better value automatically. It’s not easy to document this parameter bcs it’s hard to convey what it means. When that’s the case I’m usually more in favor of hiding those kind of details. But of course (adjusting) the resolution of the grid is smth a more keen user might be interested in. 

Hm, at the very least I’d make it bound with a lower limit (relative to the costing speed!), so to not screw up hosted servers. Maybe even including a way to opt out of it in the config generator? Maybe we should restrict total grid size instead? To make sure a user can’t ask for 200 km car isochrone with the resolution of low range ped..

Other opinions?

Reply:
Read up on it a bit, and I still agree with what I said there. It should be fairly easy to hide this kind of thing and provide a better estimate of the resolution automatically. A request option will not make sense intuitively, even to me, it’s just hard to envision how to scale this by hand. Then it’ll be a lot of trying around to micro-optimize that setting.

What do you think @akadouri? Would you be up for investigating an improved implicit approach?

Reply:
i am also uneasey about exposing this. with a really high resolution it could significantly increase the time required to mark all the pixels and then to serialize the result. 

if the primary use case is further smoothing of the results we have a couple options:

1. use the new geotiff raster output. double triple or whatever the resolution of the geotif and then apply raster based smoothing on it (eg gaussian). you'd then have to use something like gdal to contour the image to get the vectors out of it thought
2. implement other types of "simplification/generalization". right now we have the RDP algorithm implemented but there are a ton of simplification algrithms that smooth the resulting geometry, grass has a nice collection of them: https://grass.osgeo.org/grass83/manuals/v.generalize.html